runtime/gc_include/j9mm.hdf

```diff
@@ -60,5 +60,19 @@ struct J9VMThread;
 		<struct>MM_InterruptCompilationEvent</struct>
 		<data type="struct J9VMThread*" name="currentThread" description="current thread" />
 	</event>
+	
+	<event>
+		<name>J9HOOK_MM_OBJECT_ALLOCATION_SAMPLING</name>
+		<description>
+			Triggered when the object allocations sampling threshold is reached
+		</description>
+		<struct>MM_ObjectAllocationSamplingEvent</struct>
+		<data type="struct J9VMThread*" name="currentThread" description="current thread" />
+		<data type="U_64" name="timestamp" description="time of event" />
+		<data type="UDATA" name="eventid" description="unique identifier for event" />
+		<data type="omrobjectptr_t" name="object" description="the object which has been allocated." />
+		<data type="struct J9Class*" name="clazz" description="the class of the object just allocated" />
+		<data type="uintptr_t" name="objectSize" description="the size of the object just allocated" />
```

Comment:
Thanks @charliegracie , this looks good to me. Just note that this JEP doesn't require `timestamp` and `eventid`, `clazz` could be derived from `object` however it is nice to receive it since `traceAllocateObject` already has it.

Reply:
timestamp and eventID are added because the GC hooks all add them.